### ISSUE TITLE
Could I add some custom parameters to my markdown files?

### ISSUE THREAD

**Issue by zry656565** [mkdocs/mkdocs#300]
Could I add some custom parameters to my markdown files?

for example, like Jekyll

``` markdown

---
key: 10001

---

<!-- The content goes here -->
###About Cherry.js
cherry.js is ...
```

Then, I can get a parameter called "key" with value "10001" in this page.

Is there any way that I can follow? Thanks a lot!

**Comment by luispedrofonseca** [comment]
I'm trying to pass some parameters using the metadata but I'm getting a little bit confused. So, at the top of my .md file I have:

custom:hello

Then, on my template I have something like:

{{ meta.custom == 'hello' }}

Which always prints "False". If I have this:

{{ meta.custom }}

It will print:

[u'hello']


My question is, how can I compare the metadata values?

**Comment by waylan** [comment]
@luispedrofonseca If you want to compare two variables, then want to be using an [if statement][1], which is a [control structure][2] (`{% ... %}`) rather than an [expression][3] (`{{ ... }}`), which prints to the template.

[1]: http://jinja.pocoo.org/docs/dev/templates/#if
[2]: http://jinja.pocoo.org/docs/dev/templates/#list-of-control-structures
[3]: http://jinja.pocoo.org/docs/dev/templates/#expressions

**Comment by luispedrofonseca** [comment]
Yes, I know. I was just printing to see the results... The issue here is that I have this metadata on my .md file:

hide_toc: true

And then on my toc.html template when I do something like

{% if meta.hide_toc == true %}

It always evaluates to false for some reason. The metadata exists because if I print it I get:

[u'true']

Any idea what's going on?

**Comment by waylan** [comment]
Notice that the value returned is `[u'true']`, that is a list (`[]`) of length one in which the first (and only) item is a Unicode string (`u'true'`). So, no, it is not equal to `'true'`, but it is equal to `['true']`.

This is a side-effect of MkDocs using the [TOC][1] extension. As the docs for that extension state:

> Note that the keys are all lowercase and the values consist of a list of strings where each item is one line for that key. This way, one could preserve line breaks if desired. Or the items could be joined where appropriate. No assumptions are made regarding the data. It is simply passed as found to the Meta attribute.

A possible solution would be to use the docdata lib with it's [transformations][2] (or perhaps a similar homegrown solution). However, that would require internal changes to MkDocs. I expect when a plugin API becomes available, this would be possible (see #206). Until then, you'll have to work with the meta-data as a list.

    {% if meta.hide_toc[0] == true %}

[1]: https://pythonhosted.org/Markdown/extensions/meta_data.html#accessing-the-meta-data
[2]: https://github.com/waylan/docdata#transformations

**Comment by luispedrofonseca** [comment]
Thanks for the very clear explanation mate! That solved it! ;)
